shared_model/interfaces/iroha_internal/transaction_sequence.cpp

```diff
@@ -88,11 +89,11 @@ namespace shared_model {
         types::SharedTxsCollectionType result;
         auto transactions_amount = 0u;
         for (const auto &batch : batches_) {
-          transactions_amount += batch.transactions().size();
+          transactions_amount += batch->transactions().size();
         }
```

Comment:
I think, we counting size can be done more elegantly by
`auto transactions_amount = std::accumulate(
std::next(std::begin(batches_)),
std::end(batches_),
batches_->front().transactions().size(),
[](size_t acc_size, decltype(batches_->front()) batch) {
return acc_size + batch.transactions().size();
});
`
Sorry for styling, for some reason new lines did not appear